src/Types/Value.purs

```diff
@@ -388,16 +371,51 @@ instance Split Value where
     bimap (flip Value mempty) (flip Value mempty) (split coin)
       <> bimap (Value mempty) (Value mempty) (split nonAdaAsset)
 
--- FIX ME: https://github.com/Plutonomicon/cardano-transaction-lib/issues/193
--- Because our `Value` is different to Plutus, I wonder if this will become an
--- issue.
+newtype PlutusValue = PlutusValue Plutus.Value
+
+derive instance Newtype PlutusValue _
+
+instance FromPlutusType Identity PlutusValue Value where
+  fromPlutusType (PlutusValue plutusValue) =
+    Identity (adaValue <> mkValue mempty nonAdaAssets)
+    where
+    { adaTokenMap, nonAdaTokenMap } =
+      (\x -> { adaTokenMap: x.yes, nonAdaTokenMap: x.no }) <<<
+        partition (\(cs /\ _) -> cs == PlutusValue.adaSymbol) $
+        (unwrap $ PlutusValue.getValue plutusValue)
+
+    adaValue :: Value
+    adaValue = flip mkValue mempty <<< wrap <<< fromMaybe zero $ do
+      adaTokens <- Tuple.snd <$> head adaTokenMap
+      PlutusMap.lookup adaToken adaTokens
+
+    nonAdaAssets :: NonAdaAsset
+    nonAdaAssets = unsafePartial $ fromJust
+      $ mkNonAdaAssetsFromTokenMap
+      $ nonAdaTokenMap <#> \(cs /\ tokens) ->
+          PlutusValue.getCurrencySymbol cs /\ Map.fromFoldable (unwrap tokens)
+
+instance ToPlutusType Identity Value PlutusValue where
+  toPlutusType (Value (Coin adaAmount) (NonAdaAsset nonAdaAssets)) =
+    Identity $ PlutusValue (adaValue <> fold nonAdaValues)
+    where
+    adaValue :: Plutus.Value
+    adaValue
+      | adaAmount == zero = mempty
+      | otherwise = PlutusValue.lovelaceValueOf adaAmount
+
+    nonAdaValues :: Array Plutus.Value
+    nonAdaValues =
+      flip concatMap (Map.toUnfoldable nonAdaAssets) $ \(cs /\ tokens) ->
+        Map.toUnfoldable tokens <#> \(tn /\ val) ->
+          unsafePartial $ fromJust $
+            PlutusValue.singleton' (getCurrencySymbol cs) (getTokenName tn) val
+
 instance FromData Value where
-  fromData (PD.List [ coin, nonAdaAsset ]) =
-    Value <$> fromData coin <*> fromData nonAdaAsset
-  fromData _ = Nothing
+  fromData pd = (unwrap <<< fromPlutusType <<< PlutusValue) <$> fromData pd
 
 instance ToData Value where
-  toData (Value coin nonAdaAsset) = toData (coin /\ nonAdaAsset)
+  toData = toPlutusType >>> unwrap >>> unwrap >>> toData
```

Comment:
Am I correct in thinking we want to remove `ToData`/`FromData` for the CSL-style `Value`? I think it only makes sense to have it for the Plutus version? We similarly want to remove `ToData/FromData` (Purescript) Map and only have AssocMap to prevent confusion.?
I.e. remove
```
instance (ToData k, ToData v) => ToData (Map k v) where
  toData mp = Map $ entries # map (toData *** toData)
    where
    entries = Map.toUnfoldable mp :: Array (k /\ v)
```
and its FromData counterpart.